src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -776,19 +776,24 @@ static int do_map(int argc, const char *argv[])
 
 static int do_unmap()
 {
+  
+  int r = 0;
+
   int nbd = open_device(devpath.c_str());
   if (nbd < 0) {
     cerr << "rbd-nbd: failed to open device: " << devpath << std::endl;
     return nbd;
   }
 
-  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
-    cerr << "rbd-nbd: the device is not used" << std::endl;
+  r = ioctl(nbd, NBD_DISCONNECT);
+  if (r < 0) {
+      r = -errno;
+      cerr << "rbd-nbd: disconnect failed: " << cpp_strerror(-r) << std::endl;
```

Comment:
I am not sure about changing the error message. I can see reasons why it is this way right now. If open_device succeeded, but NBD_DISCONNECT failed, "the device is not used" looks like the most probable (if not the only) explanation. Is your experience different?